denguefever_tw/dengue_linebot/views.py

```diff
@@ -316,3 +316,63 @@ def suggestion_list(request):
 def gov_report_list(request):
     context = {'gov_reports': GovReport.objects.all().order_by('-report_time')}
     return render(request, 'dengue_linebot/gov_report_list.html', context)
+
+
+@login_required
+def push_msg_form(request):
+    context = {'areas': MinArea.objects.all()}
+    return render(request, 'dengue_linebot/push_msg.html', context)
+
+
+@login_required
+def push_msg_result(request):
+    areas_id = request.POST.getlist('msg_to')
+    content = request.POST['content']
+    img = request.POST['img']
+    error_msgs = list()
+    push_logs = list()
+
+    if not areas_id:
+        error_msgs.append('You do not choose any area!')
+    if not content and not img:
+        error_msgs.append('You do not write any content or image!')
+
+    if not error_msgs:
+        for area_id in areas_id:
+            users = list(LineUser.objects.filter(location=MinArea.objects.get(area_id=area_id)))
+            if users:
+                push_logs.extend(_push_msg(users=users, text=content, img=img))
+
+    return render(request, 'dengue_linebot/push_msg_result.html', {
+        'error_msgs':error_msgs,
+        'push_logs':push_logs
+    })
+
+
+def _push_msg(users, text, img):
```

Comment:
Currently, we have no better option than writing this code which would generate surprise result (i.e. error message)
Thus, we should write document for it.
Please follow [this guide](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) to write it.